internal/provider/resource_packer_channel.go

```diff
@@ -0,0 +1,381 @@
+						"fingerprint": {
+							ConflictsWith: []string{"iteration_assignment.0.id", "iteration_assignment.0.incremental_version"},
+							Description:   "The fingerprint of the iteration to assign to the channel.",
+							Type:          schema.TypeString,
+							Optional:      true,
+						},
+						"id": {
+							ConflictsWith: []string{"iteration_assignment.0.fingerprint", "iteration_assignment.0.incremental_version"},
+							Description:   "The id of the iteration to assign to the channel.",
+							Type:          schema.TypeString,
+							Optional:      true,
+						},
+						"incremental_version": {
+							ConflictsWith: []string{"iteration_assignment.0.fingerprint", "iteration_assignment.0.id"},
+							Description:   "The incremental_version of the iteration to assign to the channel.",
+							Type:          schema.TypeInt,
+							Optional:      true,
+							ValidateFunc:  validation.IntAtLeast(1),
+						},
+					},
+				},
+			},
+			// Computed Values
+			"author_id": {
+				Description: "The author of the channel.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"iteration": {
+				Description: "The iteration assigned to the channel.",
+				Type:        schema.TypeList,
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"fingerprint": {
+							Description: "The fingerprint of the iteration assigned to the channel.",
+							Type:        schema.TypeString,
+							Computed:    true,
+						},
+						"id": {
+							Description: "The id of the iteration assigned to the channel.",
+							Type:        schema.TypeString,
+							Computed:    true,
+						},
+						"incremental_version": {
+							Description: "The incremental_version of the iteration assigned to the channel.",
+							Type:        schema.TypeInt,
+							Computed:    true,
+						},
+					},
+				},
+			},
+			"organization_id": {
+				Description: "The ID of the organization this HCP Packer registry is located in.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"project_id": {
+				Description: "The ID of the project this HCP Packer registry is located in.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"created_at": {
+				Description: "Creation time of this build.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func resourcePackerChannelRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	bucketName := d.Get("bucket_name").(string)
+	client := meta.(*clients.Client)
+
+	loc := &sharedmodels.HashicorpCloudLocationLocation{
+		OrganizationID: client.Config.OrganizationID,
+		ProjectID:      client.Config.ProjectID,
+	}
+	if err := setLocationData(d, loc); err != nil {
+		return diag.FromErr(err)
+
+	}
+	// TODO (nywilken) may need to look into pagination depending on channel list sizes
+	resp, err := clients.ListBucketChannels(ctx, client, loc, bucketName)
```

Comment:
I wonder, isn't there an API endpoint to get a channel via its name? Is it because we don't have a guarantee of uniqueness for the channel name on a specific bucket?

Reply:
There is but that is a billable request as it returns the iteration and image. There is no endpoint just to get the channel without the image details. In speaking with @sylviamoss the preferred route at this time is to use the ListBucketChannel API endpoint.

Reply:
Oh OK gotcha, makes sense then